pygsti/tools/matrixtools.py

```diff
@@ -49,14 +82,13 @@ def is_hermitian(mx, tol=1e-9):
         True if mx is hermitian, otherwise False.
     """
     (m, n) = mx.shape
-    for i in range(m):
-        if abs(mx[i, i].imag) > tol: return False
-        for j in range(i + 1, n):
-            if abs(mx[i, j] - mx[j, i].conjugate()) > tol: return False
-    return True
+    if m != n:
+        return False
+    else:
+        return _np.all(_np.abs(mx - mx.T.conj()) <= tol)
```

Comment:
The old implementation used O(1) memory. This implementation uses O(mx.size) memory, but it avoids any Python for-loops. We could implement is_hermitian in Cython if we wanted to keep the memory requirement at O(1).